scripts/check_watermark.py

```diff
@@ -0,0 +1,27 @@
+"""
+Check that given files all contain the PyMC3 watermark.
+
+This is intended to be used a pre-commit hook, see `.pre-commit-config.yaml`.
+You can run it manually with `pre-commit run watermark --all`.
+"""
+
+import argparse
+from pathlib import Path
+import re
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filenames", nargs="*")
+    args = parser.parse_args()
+    for file_ in args.filenames:
+        assert (
+            re.search(
+                r"%load_ext watermark.*%watermark -n -u -v -iv -w",
+                Path(file_).read_text(),
+                flags=re.DOTALL,
+            )
+            is not None
+        ), (
+            f"Watermark not found in {file_} - please see the PyMC3 Jupyter Notebook Style guide:\n"
+            "https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style"
+        )
```

Comment:
Would it be useful to filter for Jupyter notebooks here, or doing it in `pre-commit-config.yml` is sufficient?

Reply:
Yes, in .pre-commit-config.yaml I've put
```
types: [jupyter]
```
for this hook, which ensures it only runs on notebooks

Reply:
@AlexAndorra actually, it might be possible to just do this with pygrep, then there wouldn't be a need for a separate script file...I'll get back to you on this

Reply:
Nevermind, tried `pygrep` with a negative behind, but it was _much_ slower